OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -1470,7 +1490,16 @@ static class DMOpcodeHandlers {
             if (value.TryGetValueAsString(out string refString)) {
                 if(int.TryParse(refString, out var refID))
                 {
-                    state.Push(new DreamValue(DreamObject.GetFromReferenceID(state.DreamManager, refID)));
+                    // TODO: This impl. isn't entirely correct, but refs aren't implemented like BYOND's yet
+                    var obj = DreamObject.GetFromReferenceID(state.DreamManager, refID);
+                    if (obj is null && state.DreamManager.ObjectTree.Strings.Count > refID)
```

Comment:
This leaves the possibility of a `ref()` on a string sharing an id with a DreamObject, and a `locate()` on that ref returning that DreamObject instead.